src/Main.elm

```diff
@@ -1,10 +1,418 @@
+            ]
+            (text label)
+
+
+viewFiltersRow : List (Element msg) -> Element msg
+viewFiltersRow =
+    row
+        [ Element.width Element.fill
+        , Element.spacing 8
+        ]
+
+
+viewFilters : AudienceType -> Element Msg
+viewFilters filterBy =
+    viewFiltersRow
+        (List.map
+            (viewFilter filterBy)
+            [ Audience.Authored, Audience.Shared, Audience.Curated ]
+        )
+
+
+viewLayoutColumn : List (Element msg) -> Element msg
+viewLayoutColumn =
+    column
+        [ Element.width Element.fill
+        , Element.height Element.fill
+        , Element.spacing 16
+        ]
+
+
+viewSucceed : Store -> SucceedState -> Element Msg
+viewSucceed store state =
+    viewLayoutColumn
+        [ case Store.select (makeSelector state) store of
+            Store.NotFound folderID ->
+                view404 folderID
+
+            Store.Root level ->
+                viewLevel none level
+
+            Store.Folder parentFolder level ->
+                viewLevel (viewGoUp parentFolder) level
+        , viewFilters state.filterBy
+        ]
+
+
+viewInitialising : Element msg
+viewInitialising =
+    viewLayoutColumn
+        [ el
+            (Element.Background.color (Element.rgb255 220 220 220)
+                :: stylesButton
+            )
+            (text " ")
+            |> List.repeat 10
+            |> viewLevelColumn
+        , viewFiltersRow
+            (List.map
+                (\_ ->
+                    viewFilterButton
+                        [ Element.Background.color (Element.rgb255 220 220 220)
+                        ]
+                        (text " ")
+                )
+                (List.range 0 2)
+            )
+        ]
+
+
+viewFailed : Http.Error -> Element msg
+viewFailed err =
+    case err of
+        Http.BadUrl url ->
+            text ("Wrong request url is provided: `" ++ url ++ "`")
+
+        Http.Timeout ->
+            text "Request is canceled by timeout"
+
+        Http.NetworkError ->
+            text "Poor internet connection"
+
+        Http.BadStatus status ->
+            text ("Request went back with bad status: `" ++ String.fromInt status ++ "`")
+
+        Http.BadBody message ->
+            text message
+
+
+viewLayout : Model -> Element Msg
+viewLayout model =
+    row
+        [ Element.width (Element.maximum 400 Element.fill)
+        , Element.height Element.fill
+        , Element.padding 8
+        , Element.Font.size 14
+        ]
+        [ case model of
+            Initialising _ ->
+                viewInitialising
+
+            Failed err ->
+                viewFailed err
+
+            Succeed store state ->
+                viewSucceed store state
+        ]
+
+
+view : Model -> Browser.Document Msg
+view model =
+    Browser.Document "Challenge"
+        [ Element.layout
+            [ Element.height Element.fill
+            ]
+            (viewLayout model)
+        ]
```

Comment:
I like the fact that you're using flow of the code I prefer - function that uses function is bellow that function where possible rather than other way around which seems to be generally prefered flow. But that might be personal bias.

Reply:
The code is like a story..)